From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 18 Oct 2025 11:45:31 +0200
Subject: [PATCH] HACK: install glib gir/typename files

They are supposed to be installed by glib, but that would create a
cyclic dependency, so install them here.
---
 gir/meson.build | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/gir/meson.build b/gir/meson.build
index c85f7525a20b..f02ba90d52ef 100644
--- a/gir/meson.build
+++ b/gir/meson.build
@@ -249,6 +249,8 @@ glib_gir = custom_target('gir-glib',
   output: 'GLib-2.0.gir',
   depends: [gir_giscanner_pymod, glib_gir_dep, gdump],
   depend_files: gir_giscanner_built_files,
+  install: true,
+  install_dir: girdir,
   env: g_ir_scanner_env,
   command: glib_command + [
     '--cflags-begin'] + glib_includes + extra_giscanner_cflags + [
@@ -316,6 +318,8 @@ gobject_gir = custom_target('gir-gobject',
   output: 'GObject-2.0.gir',
   depends: [glib_gir, gir_giscanner_pymod, gobject_gir_dep, gdump],
   depend_files: gir_giscanner_built_files,
+  install: true,
+  install_dir: girdir,
   env: g_ir_scanner_env,
   command: gobject_command + [
     '--include-uninstalled=' + glib_gir.full_path(),
@@ -368,6 +372,8 @@ uninstalled_gir_files += custom_target('gir-gmodule',
   output: 'GModule-2.0.gir',
   depends: [glib_gir, gir_giscanner_pymod, gmodule_gir_dep, gdump],
   depend_files: gir_giscanner_built_files,
+  install: true,
+  install_dir: girdir,
   env: g_ir_scanner_env,
   command: gmodule_command + [
     '--include-uninstalled=' + glib_gir.full_path(),
@@ -463,6 +469,8 @@ gio_gir = custom_target('gir-gio',
   output: 'Gio-2.0.gir',
   depends: [gobject_gir, gir_giscanner_pymod, gio_gir_dep, gdump],
   depend_files: gir_giscanner_built_files,
+  install: true,
+  install_dir: girdir,
   env: g_ir_scanner_env,
   command: gio_command + [
     '--include-uninstalled=' + gobject_gir.full_path(),
@@ -544,6 +552,8 @@ foreach gir: uninstalled_gir_files
     output: '@BASENAME@.typelib',
     depends: [gobject_gir, gircompiler, ],
     command: gircompiler_command,
+    install: true,
+    install_dir: typelibdir,
   )
 endforeach
 
